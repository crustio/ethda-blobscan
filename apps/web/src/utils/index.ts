export * from "./charts";
export * from "./ethereum";
export * from "./date";
export * from "./explorers";
export * from "./routes";
export * from "./search";
export * from "./number";
export * from "./text";
export * from "./tra";
export const hiddenGasElement = true;
