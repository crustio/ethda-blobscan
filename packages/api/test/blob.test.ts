import { TRPCError } from "@trpc/server";
import type { inferProcedureInput } from "@trpc/server";
import { describe, expect, it } from "vitest";

import type { TRPCContext } from "../src/context";
import { createTRPCInnerContext } from "../src/context";
import { appRouter } from "../src/root";
import type { AppRouter } from "../src/root";

describe("Blob router", async () => {
  it("should get all the blobs", async () => {
    const ctx = (await createTRPCInnerContext()) as TRPCContext;
    const caller = appRouter.createCaller(ctx);
    type Input = inferProcedureInput<AppRouter["blob"]["getAll"]>;
    const input: Input = {};

    const result = await caller.blob.getAll(input);
    expect(result).toMatchSnapshot();
  });

  it("should get all the blobs with pagination", async () => {
    const ctx = (await createTRPCInnerContext()) as TRPCContext;
    const caller = appRouter.createCaller(ctx);
    type Input = inferProcedureInput<AppRouter["blob"]["getAll"]>;
    const input: Input = {
      p: 2,
      ps: 2,
    };

    const result = await caller.blob.getAll(input);
    expect(result).toMatchSnapshot();
  });

  it("should get a blob by versioned hash", async () => {
    // TODO: setup initial fixture to also have some data on the database
    const ctx = (await createTRPCInnerContext()) as TRPCContext;
    const caller = appRouter.createCaller(ctx);
    type Input = inferProcedureInput<AppRouter["blob"]["getByVersionedHash"]>;
    const input: Input = {
      versionedHash: "blobHash004",
    };

    const result = await caller.blob.getByVersionedHash(input);
    expect(result).toMatchSnapshot();
  });
});
