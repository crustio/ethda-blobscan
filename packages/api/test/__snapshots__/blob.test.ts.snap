// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blob router > should get a blob by versioned hash 1`] = `
{
  "blobs": [
    {
      "commitment": "commitment001",
      "size": 1000,
      "versionedHash": "blobHash001",
    },
    {
      "commitment": "commitment002",
      "size": 1100,
      "versionedHash": "blobHash002",
    },
    {
      "commitment": "commitment003",
      "size": 1200,
      "versionedHash": "blobHash003",
    },
    {
      "commitment": "commitment004",
      "size": 1300,
      "versionedHash": "blobHash004",
    },
    {
      "commitment": "commitment005",
      "size": 1400,
      "versionedHash": "blobHash005",
    },
    {
      "commitment": "commitment006",
      "size": 1500,
      "versionedHash": "blobHash006",
    },
  ],
  "totalBlobs": 0,
}
`;

exports[`Blob router > should get all the blobs 1`] = `
{
  "blobs": [
    {
      "commitment": "commitment001",
      "size": 1000,
      "versionedHash": "blobHash001",
    },
    {
      "commitment": "commitment002",
      "size": 1100,
      "versionedHash": "blobHash002",
    },
    {
      "commitment": "commitment003",
      "size": 1200,
      "versionedHash": "blobHash003",
    },
    {
      "commitment": "commitment004",
      "size": 1300,
      "versionedHash": "blobHash004",
    },
    {
      "commitment": "commitment005",
      "size": 1400,
      "versionedHash": "blobHash005",
    },
    {
      "commitment": "commitment006",
      "size": 1500,
      "versionedHash": "blobHash006",
    },
  ],
  "totalBlobs": 0,
}
`;

exports[`Blob router > should get all the blobs 2`] = `
{
  "blobs": [
    {
      "commitment": "commitment001",
      "size": 1000,
      "versionedHash": "blobHash001",
    },
    {
      "commitment": "commitment002",
      "size": 1100,
      "versionedHash": "blobHash002",
    },
    {
      "commitment": "commitment003",
      "size": 1200,
      "versionedHash": "blobHash003",
    },
    {
      "commitment": "commitment004",
      "size": 1300,
      "versionedHash": "blobHash004",
    },
    {
      "commitment": "commitment005",
      "size": 1400,
      "versionedHash": "blobHash005",
    },
    {
      "commitment": "commitment006",
      "size": 1500,
      "versionedHash": "blobHash006",
    },
  ],
  "totalBlobs": 0,
}
`;

exports[`Blob router > should get all the blobs with pagination 1`] = `
{
  "blobs": [
    {
      "commitment": "commitment003",
      "size": 1200,
      "versionedHash": "blobHash003",
    },
    {
      "commitment": "commitment004",
      "size": 1300,
      "versionedHash": "blobHash004",
    },
  ],
  "totalBlobs": 0,
}
`;
